modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -50,39 +59,79 @@ const getAtlasSpecUrl = async (apiKeyword: string) => {
   return oasFileURL;
 };
 
+interface GetOASpecParams {
+  sourceType: string;
+  source: string;
+  output: string;
+  pageSlug: string;
+  repoPath: string;
+  redocExecutor: RedocExecutor;
+  apiVersion?: string;
+  resourceVersion?: string;
+}
+
+async function getOASpec({
+  source,
+  sourceType,
+  repoPath,
+  pageSlug,
+  redocExecutor,
+  output,
+  apiVersion,
+  resourceVersion,
+}: GetOASpecParams) {
+  try {
+    let spec = '';
+    const buildOptions: RedocBuildOptions = {};
+    if (sourceType === 'url') {
+      spec = source;
+    } else if (sourceType === 'local') {
+      const localFilePath = normalizePath(`${repoPath}/source/${source}`);
+      spec = localFilePath;
+    } else if (sourceType === 'atlas') {
+      spec = await getAtlasSpecUrl({ apiKeyword: source, apiVersion, resourceVersion });
+      // Ignore "incompatible types" warnings for Atlas Admin API/cloud-docs
+
+      buildOptions['ignoreIncompatibleTypes'] = true;
+    } else {
+      throw new Error(`Unsupported source type "${sourceType}" for ${pageSlug}`);
+    }
+
+    const filePathExtension = `${apiVersion ? `/v${apiVersion.split('.')[0]}` : ''}${
+      resourceVersion ? `/${resourceVersion}` : ''
```

Comment:
Should the last conditional here (like in `versionExtension` above) have two truthy checks - for `resourceVersion` and `apiVersion`? This would just ensure that we wouldn't get half of a correct filePathExtension (ie '-2020-10-02' with no apiVersion) in an odd edge case. We'd either want a fully valid filePath or an empty string, I think.

Reply:
Good catch! Thank you for that, will update accordingly.